### PR TITLE
Don't sign `BlobSidecar` and delete `BlindedBlockContents`

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description

- Stop signing `BlobSidecar`
- Delete `BlindedBlockContents` and `SignedBlindedBlockContents` and `BlindedBlockContainer`
- Changes to `BlockFactoryDeneb` and `BlockContainerSignerDeneb`

## Fixed Issue(s)
helps with builder block production flow for https://github.com/Consensys/teku/issues/7654

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: validator/client/src/test/java/tech/pegasys/teku/validator/client/duties/BlockProductionDutyTest.java [hunk 1]
```diff
@@ -240,17 +226,7 @@ public void forDeneb_shouldCreateAndPublishBlockContents() {
...
-    IntStream.range(0, signedBlobSidecars.size())
-        .forEach(
-            index -> {
-              final SignedBlobSidecarOld signedBlobSidecar = signedBlobSidecars.get(index);
-              final BlobSidecarOld unsignedBlobSidecar = unsignedBlobSidecars.get(index);
-              assertThat(signedBlobSidecar.getMessage()).isEqualTo(unsignedBlobSidecar);
-              assertThat(signedBlobSidecar.getSignature())
-                  .isEqualTo(blobSidecarsSignatures.get(unsignedBlobSidecar));
-            });
+    assertThat(signedBlobSidecars).isEmpty();
```
Comment (mehdi-aouadi) [inline validator/client/src/test/java/tech/pegasys/teku/validator/client/duties/BlockProductionDutyTest.java:229]: Shouldn't we remove this assertion as for now?

File: validator/client/src/test/java/tech/pegasys/teku/validator/client/duties/BlockProductionDutyTest.java [hunk 2]
```diff
@@ -573,17 +504,7 @@ public void forDeneb_shouldUseBlockV3ToCreateAndPublishBlockContents() {
...
```
Comment (mehdi-aouadi) [inline validator/client/src/test/java/tech/pegasys/teku/validator/client/duties/BlockProductionDutyTest.java:507]: Same here

File: validator/remote/src/integration-test/java/tech/pegasys/teku/validator/remote/typedef/OkHttpValidatorTypeDefClientTest.java [hunk 3]
```diff
@@ -110,12 +108,8 @@ void blockProductionFallbacksToNonBlindedFlowIfBlindedEndpointIsNotAvailable()
   void publishesBlindedBlockSszEncoded() throws InterruptedException {
     mockWebServer.enqueue(new MockResponse().setResponseCode(200));
 
-    final SignedBlockContainer signedBlockContainer;
-    if (specMilestone.isGreaterThanOrEqualTo(SpecMilestone.DENEB)) {
-      signedBlockContainer = dataStructureUtil.randomSignedBlindedBlockContents();
-    } else {
-      signedBlockContainer = dataStructureUtil.randomSignedBlindedBeaconBlock();
-    }
+    final SignedBlockContainer signedBlockContainer =
```
Comment (mehdi-aouadi) [inline validator/remote/src/integration-test/java/tech/pegasys/teku/validator/remote/typedef/OkHttpValidatorTypeDefClientTest.java]: Could be `SignedBeaconBlock`, to need to generalise anymore

File: validator/remote/src/integration-test/java/tech/pegasys/teku/validator/remote/typedef/OkHttpValidatorTypeDefClientTest.java [hunk 4]
```diff
@@ -131,12 +125,8 @@ void publishesBlindedBlockSszEncoded() throws InterruptedException {
   void publishesBlindedBlockJsonEncoded() throws InterruptedException, JsonProcessingException {
     mockWebServer.enqueue(new MockResponse().setResponseCode(200));
 
-    final SignedBlockContainer signedBlockContainer;
-    if (specMilestone.isGreaterThanOrEqualTo(SpecMilestone.DENEB)) {
-      signedBlockContainer = dataStructureUtil.randomSignedBlindedBlockContents();
-    } else {
-      signedBlockContainer = dataStructureUtil.randomSignedBlindedBeaconBlock();
-    }
+    final SignedBlockContainer signedBlockContainer =
```
Comment (mehdi-aouadi) [inline validator/remote/src/integration-test/java/tech/pegasys/teku/validator/remote/typedef/OkHttpValidatorTypeDefClientTest.java]: Same here

**Review by mehdi-aouadi — APPROVED** [review]
LGTM
Left some minor comments

File: validator/client/src/test/java/tech/pegasys/teku/validator/client/duties/BlockProductionDutyTest.java (hunk 2 above)
Comment (StefanBratanov) [reply validator/client/src/test/java/tech/pegasys/teku/validator/client/duties/BlockProductionDutyTest.java:507]: I just wanted to leave those there to remember to change the assertions when we remove these methods from the container.

File: validator/remote/src/integration-test/java/tech/pegasys/teku/validator/remote/typedef/OkHttpValidatorTypeDefClientTest.java (hunk 3 above)
Comment (StefanBratanov) [reply validator/remote/src/integration-test/java/tech/pegasys/teku/validator/remote/typedef/OkHttpValidatorTypeDefClientTest.java]: yup, good point

**Pull request merged**